agent/cmd/cuda-checkpoint-helper/main.cpp

```
@@ -0,0 +1,1271 @@
+  DaemonThreadShutdown &operator=(const DaemonThreadShutdown &) = delete;
+
+  ~DaemonThreadShutdown() noexcept {
+    *result_ = signal_owner_->StopAndJoinNoThrow();
+  }
+
+private:
+  daemon_protocol::ShutdownSignalOwner *signal_owner_;
+  daemon_protocol::ShutdownSignalOwner::ShutdownResult *result_;
+};
+
+bool ParsePositiveSeconds(const char *value, uint64_t *seconds_out) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)